### PR TITLE
Acceptance test cleanup/refactor

### PR DESCRIPTION
**Pull request by pglass**
## Changes proposed in this PR:
This includes some follow ups and refactoring that didn't fit into other PRs.

Overall, this is mostly code re-use between TestBasic and TestHCP.

* Create an ALB for the dev-server. This lets us use the Consul SDK to interact with both the dev-server and HCP servers.
* Extract Consul and ECS checks/validations into helpers that both TestBasic and TestHCP can use
* Extract some common Terraform into `test/acceptance/common-terraform`. This includes the ACL controller and test application, so that basic-install and hcp-install can reuse it.
* Also, add the ecs-cli to the `consul-ecs-test` image, and use that for fetching cloudwatch logs.

## How I've tested this PR:
Acceptance tests

## How I expect reviewers to test this PR:
👀 

## Checklist:
- [x] Permissions update to allow use of ALBs: https://github.com/hashicorp/consul-test-environments/pull/95
- [x] Tests added
- [x] ~CHANGELOG entry added~ n/a, test-only change

### CODE REVIEW

File: test/acceptance/framework/helpers/consul.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/hashicorp/terraform-aws-consul-ecs/test/acceptance/framework/logger"
+	"github.com/stretchr/testify/require"
+)
+
+// WaitForConsulServices waits for services to show in the consul service catalog.
+func WaitForConsulServices(t *testing.T, consulClient *api.Client, serviceNames ...string) {
```
Comment (pglass) [inline test/acceptance/framework/helpers/consul.go:14]: These are all "naive" refactorings, and are a little limited (can't customize timeouts). I'm open to rethinking the interfaces here... Maybe a struct + methods that can be overridden, like

```go
type TestHelpers struct {
    t *testing.T
    suite *config.TestConfig
    consulClient *api.Client
    timer *retry.Timer
}

func NewTestHelpers(...) {
    return &TestHelpers{..., timer: nil}
}

func (h *TestHelpers) WithTimer(timer *retry.Timer) *TestHelpers {
    return &TestHelpers{...., timer: timer}
}

func (h *TestHelpers) WaitForConsulServices(serviceName ...string) {
    timer := h.timer
    if timer == nil {
         // Use default timer
        timer = &retry.Timer{Timeout: 6 * time.Minute, Wait: 20 * time.Second}
    }
    ...
}
```

File: test/acceptance/framework/helpers/cloudwatch.go [hunk 2]
```diff
@@ -1,96 +1,84 @@
...
-		var resp listLogEventsResponse
-		getLogEventsOut, err := shell.RunCommandAndGetOutputE(t, shell.Command{Command: args[0], Args: args[1:]})
-		if err != nil {
-			return resp, err
-		}
-
-		err = json.Unmarshal([]byte(getLogEventsOut), &resp)
-		return resp, err
+// GetCloudWatchLogEvents fetches all log events for the given container.
+func GetCloudWatchLogEvents(t terratestTesting.TestingT, testConfig *config.TestConfig, taskId, containerName string) (LogMessages, error) {
```
Comment (erichaberkorn) [inline test/acceptance/framework/helpers/cloudwatch.go:17]: This is awesome

**Review by erichaberkorn — APPROVED** [review]
This diff is really big so it's hard to review every line, but this is amazing. The tests are much more readable after this. Great work 👍

**Review by ishustava — APPROVED** [review]
Looks great!

**Comment by pglass** [comment]
I'll split up these changes into smaller increments. Closing!

**Pull request closed without merging**